src/utils/record-completeness/ComputeRecordCompleteness.ts

```diff
@@ -1,92 +1,113 @@
-    nonCoreMandatorySchema = nonCoreMandatoryValidation.psvNonCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.TRL) {
-    techRecord = {...techRecord, primaryVrm: techRecordWrapper.primaryVrm} as any;
-    coreMandatorySchema = coreMandatoryValidation.trlCoreMandatorySchema;
-    nonCoreMandatorySchema = nonCoreMandatoryValidation.trlNonCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.LGV) {
-    coreMandatorySchema = coreMandatoryValidation.lgvCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.CAR) {
-    coreMandatorySchema = coreMandatoryValidation.carCoreMandatorySchema;
-  } else if (vehicleType === VEHICLE_TYPE.MOTORCYCLE) {
-    coreMandatorySchema = coreMandatoryValidation.motorcycleCoreMandatorySchema;
+function validateVehicleAttributes(vehicle: Vehicle): ValidationResult | undefined {
+  const vehicleType = vehicle.techRecord[0].vehicleType;
+
+  const validationSchema = vehicleType === VEHICLE_TYPE.TRL
+    ? trlCoreMandatoryVehicleAttributes
+    : psvHgvCarLgvMotoCoreMandatoryVehicleAttributes
+
+  if (!vehicleType) {
+    return missingVehicleTypeValidationResult;
+  } else if (!validationSchema) {
+    return undefined;
   } else {
-    throw new HTTPError(400, ERRORS.VEHICLE_TYPE_ERROR);
+    const vehicleAttributes = {
+      systemNumber: vehicle.systemNumber,
+      vin: vehicle.vin,
+      primaryVrm: vehicle.primaryVrm,
+      trailerId: (vehicle as Trailer).trailerId ?? undefined
+    };
+
+    return validationSchema.validate(vehicleAttributes, { stripUnknown: true });
   }
-  coreMandatoryValidationResult = validateRecordCompleteness(coreMandatorySchema, techRecord);
-  nonCoreMandatoryValidationResult = validateRecordCompleteness(nonCoreMandatorySchema, techRecord);
-  return {coreMandatoryValidationResult, nonCoreMandatoryValidationResult};
 };
 
-export const computeRecordCompleteness = (techRecordWrapper: Vehicle, trailerId?: string): string => {
-  let recordCompleteness = RECORD_COMPLETENESS_ENUM.COMPLETE;
-  let isCoreMandatoryValid = true;
-  let isNonCoreMandatoryValid = true;
-  if (!techRecordWrapper.systemNumber) {
-    throw new HTTPError(400, ERRORS.SYSTEM_NUMBER_GENERATION_FAILED);
+function validateMandatoryTechRecordAttributes(vehicle: Vehicle) {
+  const vehicleType = vehicle.techRecord[0].vehicleType as VEHICLE_TYPE;
+  
+  const coreMandatorySchema: ObjectSchema | undefined = coreMandatorySchemaMap.get(vehicleType);
+
+  if (!coreMandatorySchema) {
+    throw new HTTPError(400, ERRORS.VEHICLE_TYPE_ERROR);
   }
-  const generalAttributes = {
-    systemNumber: techRecordWrapper.systemNumber,
-    vin: techRecordWrapper.vin,
-    primaryVrm: techRecordWrapper.primaryVrm,
-    // FIXME: handle trailerId in a better way
-    trailerId
+
+  const nonCoreMandatorySchema: ObjectSchema | undefined = nonCoreMandatorySchemaMap.get(vehicleType);
+
+  const techRecord = vehicleType === VEHICLE_TYPE.TRL
+    ? { ...vehicle.techRecord[0], primaryVrm: vehicle.primaryVrm }
```

Comment:
I'm a bit confused by this logic as a TRL shouldn't have primaryVrm populated

Reply:
I will go with your knowledge and remove `primaryVrm` for trailers. This was in the code we inherited so I kept it too.